[ENH] Inverse Gaussian distribution

Omswastik-11 opened the pull request:
### Summary
This PR adds the `InverseGaussian` distribution class to `skpro.distributions`. This implementation wraps `scipy.stats.invgauss` while ensuring proper parameter mapping to match standard parameterizations (specifically `numpy.random.wald`).

### Key Features & Formulation
* **Parameterization:** Uses Mean $\mu$ and shape parameter $\lambda$ (scale).
* **Formula:**
    $$f(x; \nu, \lambda) = \sqrt{\frac{\lambda}{2\pi x^3}} \exp\left(-\frac{\lambda(x-\nu)^2}{2\nu^2 x}\right)$$
* **Correspondence:** Equivalent to `numpy.random.wald(mean=mu, scale=λ)`.
* **Broadcasting:** Full support for array-valued parameters.

### Implementation Details
* **Backend:** Wraps `scipy.stats.invgauss` with specific parameter mapping:
    * `mu = mu`
    * `scale = λ`
* **Capabilities:** Implements exact mean, variance, PDF, CDF, and PPF. Includes approximate energy and PDF normalization.
* **Files Modified:**
    * `inversegaussian.py`: Added new distribution class.
    * `__init__.py`: Exported the new class.

fkiraly commented:
Code quality checks are failing, see https://www.sktime.net/en/stable/developer_guide/coding_standards.html

Omswastik-11 commented:
<img width="962" height="757" alt="image" src="https://github.com/user-attachments/assets/767bdb57-b6f7-4d28-8a6f-80bf1775b708" />


I have implemented the requested changes for `InverseGaussian` in `inversegaussian.py`.

### Summary of Changes

* **Parameter Renaming**: Renamed `scale` $\to$ `lambda_`.
    * *Rationale*: Addressed the feedback that this is not a scale parameter. I kept `mu` as `mu` because `mean` conflicts with the inherited `.mean()` method, and `shape` conflicts with the `.shape` property. `lambda_` is used to represent $\lambda$ while avoiding Python keyword conflicts.
* **Documentation**:
    * Updated docstrings to use double backticks .
    * Updated descriptions, examples, and Scipy/Numpy mappings to reflect the change to `lambda_`.
* **Implementation**: Updated `__init__`, `_get_scipy_param`, and `get_test_params`.

**Verification**: Ran `pytest` for InverseGaussian and all tests passed successfully.

<img width="1222" height="467" alt="image" src="https://github.com/user-attachments/assets/630d1715-017f-4d6d-b12b-0c3318081e63" />

<img width="1257" height="646" alt="image" src="https://github.com/user-attachments/assets/d1f8326c-cb0b-44a3-84c3-52a249e3f092" />

fkiraly commented:
Yes, I would say, use the `numpy` convention, which has `scale`.

Omswastik-11 commented:
hi @fkiraly !! renamed `lam` to `scale` .

fkiraly reviewed (changes requested):
Great! Looks fine now regarding code, but I think the mathematical formula is now incorrect. If `scale` is a multiplicative scale parameter, then you should divide and not multiply by `scale`.

While you are at it, I would use greek symbols $\mu$ and $\sigma$ in the formula rather than writing out "scale" etc, that makes it easier to read.

Omswastik-11 commented:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.invgauss.html

This time I used Scipy's formula

fkiraly reviewed (changes requested):
Thanks, but the formula is not directly applicable. Could you not just give a formula in terms of a $\mu$ = `mu` and $\sigma$ = `scale`?

Omswastik-11 commented:
I think It would look like this if we transform scipy's eqn

<img width="315" height="67" alt="image" src="https://github.com/user-attachments/assets/71c08450-64c3-4051-a5b5-fbac11270a69" />
